src/Quadmath.jl

```diff
@@ -67,12 +72,95 @@ elseif Sys.iswindows()
     primitive type Float128 <: AbstractFloat 128
     end
     const Cfloat128 = Float128
+
+    # This is used to force alignment of subsequent Float128 structs
+    # on the call stack. Let's hope nobody notices the overhead before
+    # the Julia compiler handles fp128 itself.
+    if Sys.WORD_SIZE == 32
+        struct PadPtr{T}
+            p::Ref{T}
+            d1::Int32
+            d2::Int32
+            d3::Int32
+        end
```

Comment:
What exactly is this doing? You need to pad the call with a 128-bit aligned structure?

Reply:
Yes. In Win32, the Float128 arguments are pushed onto the stack, and the callee expects them to be aligned. This structure is used when a 32-bit pointer is pushed first, which typically happens when such a pointer is used to handle the return value. I wish I could point to documentation for this, but I wasn't able to find any; I'm basing the code here on disassembled binaries and experiments.

Reply:
Should the pointer be 128-bit aligned? Or the value be 128-bit aligned?

If it's the latter, we could probably use the same "Tuple alias" trick as I used for unix values?

Reply:
(Also, sorry for the radio silence: I've been busy the last few weeks and haven't had time to follow up on this)

Reply:
The pointer gets aligned by the ABI; the values are the tricky part.  So you may be right about the Tuple trick; I would have to experiment, but it looks like you are doing that yourself

Reply:
Okay, I think I've got this to work. Thanks for your help on this.